From a043b24029f5050013abc61c9cd5f49cb1efcef2 Mon Sep 17 00:00:00 2001
From: Guidix <guidixm@gmail.com>
Date: Wed, 17 Jul 2024 05:04:28 +0300
Subject: [PATCH] SystemUI: Set default screen res and framerate

---
 .../screenrecord/ScreenMediaRecorder.java     | 19 +++++++------------
 1 file changed, 7 insertions(+), 12 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/screenrecord/ScreenMediaRecorder.java b/packages/SystemUI/src/com/android/systemui/screenrecord/ScreenMediaRecorder.java
index 46e30282a218..b60828199481 100644
--- a/packages/SystemUI/src/com/android/systemui/screenrecord/ScreenMediaRecorder.java
+++ b/packages/SystemUI/src/com/android/systemui/screenrecord/ScreenMediaRecorder.java
@@ -72,11 +72,12 @@ import java.util.List;
  */
 public class ScreenMediaRecorder extends MediaProjection.Callback {
     private static final int TOTAL_NUM_TRACKS = 1;
-    private static final int VIDEO_FRAME_RATE = 30;
+    private static final int VIDEO_FRAME_RATE = 60;
     private static final int VIDEO_FRAME_RATE_TO_RESOLUTION_RATIO = 6;
     private static final int LOW_VIDEO_FRAME_RATE_TO_RESOLUTION_RATIO = 2;
-    private static final int LOW_VIDEO_FRAME_RATE = 25;
+    private static final int LOW_VIDEO_FRAME_RATE = 30;
     private static final int AUDIO_BIT_RATE = 196000;
+    private static final int LOW_VIDEO_BIT_RATE = 15000000;
     private static final int AUDIO_SAMPLE_RATE = 44100;
     private static final int MAX_DURATION_MS = 60 * 60 * 1000;
     private static final long MAX_FILESIZE_BYTES = 5000000000L;
@@ -164,20 +165,14 @@ public class ScreenMediaRecorder extends MediaProjection.Callback {
 
         mMediaRecorder.setOutputFormat(MediaRecorder.OutputFormat.MPEG_4);
 
-
         // Set up video
         DisplayMetrics metrics = new DisplayMetrics();
         WindowManager wm = (WindowManager) mContext.getSystemService(Context.WINDOW_SERVICE);
         wm.getDefaultDisplay().getRealMetrics(metrics);
-        int refreshRate = mLowQuality ? LOW_VIDEO_FRAME_RATE :
-                (int) wm.getDefaultDisplay().getRefreshRate();
-        if (mMaxRefreshRate != 0 && refreshRate > mMaxRefreshRate) refreshRate = mMaxRefreshRate;
-        int[] dimens = getSupportedSize(metrics.widthPixels, metrics.heightPixels, refreshRate);
-        int width = dimens[0];
-        int height = dimens[1];
-        refreshRate = dimens[2];
-        int resRatio = mLowQuality ? LOW_VIDEO_FRAME_RATE_TO_RESOLUTION_RATIO
-                : VIDEO_FRAME_RATE_TO_RESOLUTION_RATIO;
+        int width = 720;
+        int height = 1600;
+        int refreshRate = 60;
+        int resRatio = mLowQuality ? LOW_VIDEO_FRAME_RATE_TO_RESOLUTION_RATIO : VIDEO_FRAME_RATE_TO_RESOLUTION_RATIO;
         int vidBitRate = width * height * refreshRate / VIDEO_FRAME_RATE * resRatio;
         long maxFilesize = mLongerDuration ? MAX_FILESIZE_BYTES_LONGER : MAX_FILESIZE_BYTES;
         if (!mHEVC) {
-- 
2.44.0.windows.1

